### ISSUE TITLE
Most recent version of Wire crashes upon start

### ISSUE THREAD

**Issue by 0ki** [wireapp/wire-ios#3921]
## Checklist (please check before submitting)

* [ x ] I checked the existing issues, and the issue was not reported before.

## The problem

Briefly describe the issue you are experiencing (or the feature you want to see added to Wire).
Tell us what you were trying to do and what happened instead.

## Environment

* Wire version that exhibits the issue: 3.49
* Last Wire version that did not exhibit the issue (if applicable): 3.48
* iOS version used to run Wire: 13.3
* iOS device type: iPhone SE
* Mobile network type (EDGE/LTE/Wi-Fi/Offline): any
* Exact date, time and timezone (only for the crash issues): at the moment of submitting this post and 15 minutes before (7JAN 06:50 GMT+2 ...)

## Details

After updating to the newest version, wire was loading for 5-10 seconds (logo) and then crashed. Now it consistently crashes immediately upon opening. I have three different accounts logged in, all without a phone number.

**Comment by 0ki** [comment]
If that changes anything: I've been using the accounts for a very long time, and there've been timed out logouts in the past. Most recent one I just re-logined a couple of days ago.

One of the accounts might be in the logged out state right now as well.


Advice?

**Comment by typfel** [comment]
Does it also crash if you try to open the app while in air plane mode, that would indicate that crash is triggered by event processing.

**Comment by 0ki** [comment]
Yes, it does.

**Comment by typfel** [comment]
Ok, so to investigate this further we would need crash logs. You can retrieve them from:

 Settings > Privacy > Analytics & Improvements  > Analytics Data

In that list there should a recent entry from the Wire app which you could then share with us.

**Comment by 0ki** [comment]
The list is there.
The most recent entry from Wire.wakeups_resouce is July 2019.
Wire.cpu_resource - August 2019.
Wire - July 2019.
Wire Share Extension - January 2020. I had the idea to try share extension out to get the list of my recent chats (which is the minimum I need to desperately get out of the app) but that didn't work either.


Which one do you need and how do I send it to you securely?

**Comment by typfel** [comment]
I would be interested in one named like:

`Wire-2020-01-07-xxxxxx.ips.synced`

If you can't find such log it's probably because the there's a limit on how many logs will be stored. Then you need perform the following steps:

1. Connect your iPhone to your Mac and sync it with iTunes (This will delete the analytics logs from the phone)
2. Start the Wire  app to reproduce the crash again
3. Go to Analytics Data and look for `Wire-2020-01-08-xxxxxx.ips.synced`

You can send the log either by connecting to me at Wire @jacobp or to someone else at Wire and then tell them to forward the logs to me.

**Comment by 0ki** [comment]
Thanks for the thorough instructions. This worked.
Only last one or two files are actual crashlogs from me launching the app. The others just starting appearing there automatically as soon as I cleared the old logs about twice per minute. I guess those are push initiated? Also explains my poor battery life for the past two days.

hxxp://02.lv/f/2020/01/08/stuff.7z

**Comment by 0ki** [comment]
Thank you for tagging the 3.49 release on github. I'm reviewing the source code and looking at possible problems now.
Some peculiarities about my account:
* I have chats with users that no longer have an account. Showing up as "..."
* I have groups that have such users.
* I have groups where I'm the only participants.
* I have outgoing connection requests with users that don't have a username.
* I have outgoing connection request with a user that has deleted their username before confirming my request. This chat was archived. (^1)

Some non-related crashes I've found:
* Searching for a user that starts with a symbol (e.g. "." or "%") crashes the app.
* Unarchiving and opening the chat (^1) shows up as a group with no participants but is a chat with myself on iOS where I can't send any new messages. When I click on my username on the top of the screen, app crashes. (On web this shows up a chat with "..." and I can send messages.) I can't delete this "chat" or "cancel this request".

**Comment by typfel** [comment]
Thanks! The crash logs were helpful. It's crashing due to force unwrapping of an optional during the database migration.

Should hopefully be easy to fix and available in an upcoming test flight build.

**Comment by 0ki** [comment]
Great. I'm eagerly waiting for the build. Do you have a link to an article describing setting up wire test flight build on my iPhone? 

Will switching from 3.49 to test flight and later back to 3.50 preserve my chat logs and keys?

**Comment by typfel** [comment]
Yes, you will retain your chat logs and keys but you need to be careful when you stop using the test flight build, because downgrading to an earlier version may potentially drop the database. I would recommend to only go back to the App Store version after it has caught up with your test flight build.

You can become a test flight user by following this link https://testflight.apple.com/join/TiFZsCXi

**Comment by 0ki** [comment]
OK. Can I already install the app in testflight or is the fix not out yet?

Please ping me, when I can.

**Comment by typfel** [comment]
The fix is not out yet. I'll write here when it's out.

**Comment by typfel** [comment]
Build 3.50 (3985) is out on testflight with a potential fix.

**Comment by 0ki** [comment]
That was worth the wait. Fixed. Thanks!

I've also now exported the conversation history backup. Does it contain media as well? iOS is saying that wire is using 500MB for it's data, but the backup file is only 50MB. Is there a tool to decrypt/view the backup for the next time when a wire update kills everything? If not, I suppose I can write my own using wire-ios-cryptobox/WireCryptobox/ChaCha20Encryption.swift
